### PR TITLE
Migration for all users table

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Migration for users, users-profile, student, teacher and roles.
-  Changed the EUserRole from admin and super_user to student and teacher.
- Changed the seeder file, user controller and user service as there were errors for the previous roles.

## Associated Ticket Link(s)
- https://trello.com/c/DlhtaTgt

## Screenshots / Screen Recordings:
N/A

### CODE REVIEW

File: src/common/entities/student.entitiy.ts
Comment (ragib-hossain-sazim) [inline src/common/entities/student.entitiy.ts:1]: `student.entitiy.ts` => `students.entity.ts`

File: src/common/entities/student.entitiy.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  @OneToOne(() => UserProfile)
+  userProfile!: UserProfile;
+
+  @Enum(() => EducationLevel)
+  @Property({ fieldName: 'education_level' })
+  educationLevel!: EducationLevel;
+
+  @Enum(() => Medium)
+  @Property({ fieldName: 'medium', nullable: true })
```
Comment (ragib-hossain-sazim) [inline src/common/entities/student.entitiy.ts:21]: [JC] why are you making all these fields optional?

File: src/common/enums/educationLevel.enum.ts
Comment (ragib-hossain-sazim) [inline src/common/enums/educationLevel.enum.ts:1]: please follow the naming convention of the files don't use camelCase. This should be `education-levels.enum.ts`.

File: src/common/enums/studentMedium.enum.ts
Comment (ragib-hossain-sazim) [inline src/common/enums/studentMedium.enum.ts:1]: Also I don't think you need to create seperate files for all enums. Looks these enums are associated with students entity. You should create a file called `students.enum.ts` and then put and export all the related enums from there.

For your case these are => `EEducationLevel`, `EMedium`, `EDegree`

File: src/common/enums/userGender.enum.ts
Comment (ragib-hossain-sazim) [inline src/common/enums/userGender.enum.ts:1]: this can go under `users.enum.ts`

File: src/common/enums/studentMedium.enum.ts
Comment (ghost) [reply src/common/enums/studentMedium.enum.ts:1]: Bhaiya, I am closing this PR. As I had to do some significant changes for migration.

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ import { Body, Controller, Get, Post, UseGuards, UseInterceptors } from "@nestjs
 
 import { ITokenizedUser } from "@/auth/auth.interfaces";
 import { CurrentUser } from "@/auth/decorators/current-user.decorator";
-import { Roles } from "@/auth/decorators/roles.decorator";
+// import { Roles } from "@/auth/decorators/roles.decorator";
```
Comment (ragib-hossain-sazim) [inline src/users/users.controller.ts:5]: Please remove commented out codes

**Pull request closed without merging**